SPECS/mariner-rpm-macros/macros

```diff
@@ -1,29 +1,17 @@
-#
-#   System settings
-#
-%_topdir            /usr/src/mariner
-#
-#   Some standard locations
-#
-%_sysconfdir        /etc
-%_lib               lib
-%_lib64             lib64
-%_prefix            /usr
-%_exec_prefix       %{_prefix}
-%_bindir            %{_exec_prefix}/bin
-%_sbindir           %{_exec_prefix}/sbin
-%_libdir            /usr/%{_lib}
-%_lib64dir          /usr/%{_lib64}
-%_libexecdir        %{_exec_prefix}/libexec
-%_datadir           %{_prefix}/share
-%_docdir            %{_datadir}/doc
-%_includedir        %{_prefix}/include
-%_infodir           %{_datadir}/info
-%_mandir            %{_datadir}/man
-%_oldincludedir     /usr/include
-%_sharedstatedir    /var/lib
-%_var               /var
-%_localstatedir     /var
+
+# RPM moved away from the "system" _topdir over 16 years ago:
+# https://github.com/rpm-software-management/rpm/commit/15f33b33e62a703105671dff675689265da88b16
+# Fedora/RHEL have long since adopted the move, while SUSE does have a
+# fallback that causes root (and/or anyone with write access) to use
+# /usr/src/packages, which was the original "system" _topdir. For
+# obvious reasons, it's important for non-root users to be able to
+# build packages (in fact root should generally NEVER be used to build
+# any package), but there may be lots of external code that has
+# hardcoded the /usr/src/mariner directory for building. So this
+# adopts SUSE's approach, while using our /usr/src/mariner directory
+# instead of the traditional /usr/src/packages.
+%_topdir            %{expand:%%global _topdir %{lua:if posix.access(rpm.expand("/usr/src/mariner"), "w") then print "/usr/src/mariner" else print "%{getenv:HOME}/rpmbuild" end} \
```

Comment:
We will want to do a few tests on this one, make sure it works as expected. I can try some local tools if you queue a full end-to-end build (I'd turn on the raw toolchain boostrap as well, since that might have some knock on effects).

Reply:
here's pipeline run:
https://dev.azure.com/mariner-org/mariner/_build/results?buildId=516244&view=results

Reply:
new pipeline build with latest push https://dev.azure.com/mariner-org/mariner/_build/results?buildId=516306&view=results

Reply:
Found a few spots its hard-coded in the tools:
`build_mariner_toolchain.sh`, `build_official_toolchain_rpms.sh`, `toolchain.mk` (aside these should probably just be inheriting from each other if they truly need to be hard-coded...) @anphel31

Also the containerized build env. will need a polish up @neha170 .

I'm supportive of getting this change in as well, but at minimum the toolchain bits will need adjusting.

Reply:
https://dev.azure.com/mariner-org/mariner/_build/results?buildId=516389&view=results

Reply:
I updated all the places in the toolkit that grep showed it was hardcoded and tried to fix them (but @anphel31 and @neha170 should review it).

And also as you said above, we probably shouldn't be hardcoding rpm build locations into the toolkit, at least not in multiple different places.